app/components/restoration_game.rb

```diff
@@ -0,0 +1,163 @@
+class RestorationGame < ViewComponent::Base
+  include Motion::Component
+  include RestoConstants
+  include RestorationInitialize
+  include WorldGeneration
+
+  attr_reader :board, :board_viewed, :index, :selected, :board_size, :view_size, :view_corner, :time_passed, :zoom
+  attr_reader :berries, :seeds, :saplings, :water, :water_used
+  attr_reader :saplings_to_give, :seeds_to_give
+  attr_reader :show_info, :show_intro_msg, :show_win_msg, :won
+
+  map_motion :change_selected
+  map_motion :close_intro_msg
+  map_motion :close_win_msg
+  map_motion :paint
+  map_motion :pan
+  map_motion :toggle_info_msg
+  every 1.second, :update
+
+  def initialize
```

Comment:
I think there's some refactoring that could be done in this class move some of this logic into some Tile Type subclasses. For example, Water tiles could know how much health they grant and how they get evaluated.